### PR TITLE
Kanderson/us109627 assignment type

### PR DESCRIPTION
**Pull request by kieranderson**
<a href="https://rally1.rallydev.com/#/detail/userstory/331297385828?fdp=true">US109627</a>

This PR adds the ability to see and set the assignment type for an assignment between individual and group

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+				? this.localize('txtIndividual')
+				: this.localize('txtGroup');
+
+		if (!this._isIndividualType) {
+			const selectedGroupCategoryName = assignment.getAssignmentTypeSelectedGroupCategoryName();
+			this._groupTypeText = this.localize('txtGroupCategoryWithName', 'groupCategory', selectedGroupCategoryName);
+		}
+	}
+
+	_getGroupCategoryOptions() {
```
Comment (kieranderson) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js]: The formatting of this looks odd, linting made me do this.

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+				.d2l-body-small {
+					margin: 0.5rem 0 0.3rem 0;
+				}
+
+				.d2l-body-compact {
+					margin: 0 0 0.3rem 0;
+				}
+
+				.group-info {
+					padding-left: 35px;
```
Comment (ppaskaris-d2l) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js]: Do we do pixels or rems for this kind of stuff? TODO: look this up

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+		super();
+		this._setEntityType(AssignmentEntity);
+		this._folderTypeText = '';
+		this._groupCategories = [];
+		this._groupHomepageLink = '';
+		this._groupTypeDisabled = false;
+		this._groupTypeText = '';
+		this._infoText = '';
+		this._isIndividualType = false;
+		this._isReadOnly = '';
```
Comment (ppaskaris-d2l) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js]: Should this be initialized to a bool?

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js (hunk 1 above)
Comment (ppaskaris-d2l) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js]: Thanks, I hate it. Perhaps try this instead..?

```js
const options = this._groupCategories.map(
        option => html`
            <option value=${option.value} ?selected=${option.selected}>
                ${option.title}
            </option>
            `
    );

return html`${options}`;
```

Or don't, it doesn't super matter to me.

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+			selectStyles,
+			css`
+				:host {
+					display: block;
+				}
+				:host([hidden]) {
+					display: none;
+				}
+
+				block-select {
```
Comment (ppaskaris-d2l) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js]: This is a tag selector, but it should be a class selector `.block-select { ... }`

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js (hunk 2 above)
Comment (kieranderson) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-type-editor.js]: Looks like REM

**Pull request merged**